### ISSUE TITLE
SnapshotGenerator - support implicit type constraints on renamed elements

### ISSUE THREAD

**Issue by wmrutten** [FirelyTeam/firely-net-sdk#1074]
FHIR allows a shortcut for renamed choice types, e.g. from BodyHeight profile:

```
<element id="Observation.valueQuantity">
   <path value="Observation.valueQuantity"/> 
</element>
```

The element renaming implies a type constraint on Quantity:

```
<type> 
   <code value="Quantity"/> 
</type> 
```

The `SnapshotGenerator` class should support this shortcut syntax.

**Comment by wmrutten** [comment]
Also generate predefined slice name, if not specified, from new element name:

```
<sliceName value="valueQuantity"/> 
```

**Comment by mmsmits** [comment]
fixed through #1089
